scalableminds/webknossos
# React Router

This PR is focused on the frontend changes necessary for the new security handling through Silhouette. Views for Login / Registration etc are now handled by the single page app as well, further eliminating server side views.

### Changes:
- Added React Router
- Switched many <a> tags to <Link> to place nice with React Router
- New reactive Navbar
- Moved the currently active user information into the store. No more `app.currentUser`
- Replaced server-side views for FAQ and Keyboard Controls with React views
- Reduced  initial server-side page skeleton to an minimum
- (Reactified view for Login / Registration etc are part of the silhouette branch already)


### Mailable description of changes:
- Added new nav bar
- Changed the internals for URL handling

### Steps to test:
**DO NOT TEST ANY SECURITY FEATURES YET** We can test these separately once the whole Silhouette branch is done.
- Monkey Test every Link. Almost all anchor tags were replaced so we should click on every view, button etc...
- Make sure there is a warning when leaving any tracing mode without having saved first

------
